src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -68,50 +71,53 @@ public New3Command(string commandName, ITemplateEngineHost host, ITelemetryLogge
+        /// <param name="telemetryLogger"><see cref="ITelemetryLogger"/> to use to track events.</param>
+        /// <param name="onFirstRun">actions to be run on the first run.</param>
+        /// <param name="args">arguments to be run using template engine.</param>
+        /// <param name="hivePath">(optional) the path to template engine settings to use.</param>
+        /// <returns>exit code: 0 on success, other on error.</returns>
+        /// <exception cref="CommandParserException">when <paramref name="args"/> cannot be parsed.</exception>
+        public static int Run(string commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, Action<IEngineEnvironmentSettings> onFirstRun, string[] args, string? hivePath = null)
         {
-            return Run(commandName, host, telemetryLogger, new New3Callbacks() { OnFirstRun = onFirstRun }, args, null);
+            return Run(commandName, host, telemetryLogger, new New3Callbacks() { OnFirstRun = onFirstRun }, args, hivePath);
         }
 
-        private static readonly Guid _entryMutexGuid = new Guid("5CB26FD1-32DB-4F4C-B3DC-49CFD61633D2");
-        private static Mutex? _entryMutex;
-
         private static Mutex EnsureEntryMutex(string? hivePath, ITemplateEngineHost host)
         {
             if (_entryMutex == null)
             {
-                string _entryMutexIdentity;
+                string entryMutexIdentity;
 
                 // this effectively mimics EngineEnvironmentSettings.BaseDir, which is not initialized when this is needed.
                 if (!string.IsNullOrEmpty(hivePath))
                 {
-                    _entryMutexIdentity = $"{_entryMutexGuid.ToString()}-{hivePath}".Replace("\\", "_").Replace("/", "_");
+                    entryMutexIdentity = $"{_entryMutexGuid.ToString()}-{hivePath}".Replace("\\", "_").Replace("/", "_");
                 }
                 else
                 {
-                    _entryMutexIdentity = $"{_entryMutexGuid.ToString()}-{host.HostIdentifier}-{host.Version}".Replace("\\", "_").Replace("/", "_");
+                    entryMutexIdentity = $"{_entryMutexGuid.ToString()}-{host.HostIdentifier}-{host.Version}".Replace("\\", "_").Replace("/", "_");
                 }
 
-                _entryMutex = new Mutex(false, _entryMutexIdentity);
+                _entryMutex = new Mutex(false, entryMutexIdentity);
             }
 
             return _entryMutex;
         }
 
-        public static int Run(string commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, Action<IEngineEnvironmentSettings> onFirstRun, string[] args, string? hivePath)
-        {
-            return Run(commandName, host, telemetryLogger, new New3Callbacks() { OnFirstRun = onFirstRun }, args, hivePath);
-        }
-
-        public static int Run(string commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, New3Callbacks callbacks, string[] args, string? hivePath)
```

Comment:
Callback with `New3Callbacks` is now private, it should be public

Reply:
should we remove other overloads then? I see no sense keeping them, as the only caller of it should be SDK.

Reply:
I'm fine with removing other overloads

Reply:
Removed other one.